2020/webkitgtk-npapi-sunsetting.markdown

```diff
@@ -0,0 +1,195 @@
+  [removed
+  support](/2019/webkit-2.26-six-month-retrospective.html#cleanups) for NPAPI
+  plug-ins which used GTK2, but the rest of plug-ins kept working.
+* WebKitGTK 2.30.x will be the last stable series with support for NPAPI
+  plug-ins _at all_. Version [2.30.0 was
+  released](https://webkitgtk.org/2020/09/11/webkitgtk2.30.0-released.html) a
+  couple of weeks ago.
+* WebKitGTK 2.32.0, due in March 2021, will be the first *stable* release
+  to ship without support for NPAPI plug-ins.
+* We have already [removed the relevant
+  code](https://trac.webkit.org/changeset/265753/) from the WebKit repository.
+* While the WPE WebKit port allowed running windowless NPAPI plug-ins, this
+  was never advertised nor supported by us.
+
+
+## What is NPAPI?
+
+NPAPI, short for _Netscape Plugin Application Programming Interface_ allowed
+browser plug-ins to extend the functionality of a web browser providing
+support for additional content types. NPAPI appeared first in Netscape
+Navigator 2.0, in 1995.
+
+When a NPAPI plug-in is used to render content, the web browser will
+effectively _carve a hole_ in the rectangular location where content handled
+by the plug-in will be placed, and hands off the rendering responsibility to
+the plug-in.
+
+## What is NPAPI used for?
+
+A number of technologies have used NPAPI along the years for different
+purposes:
+
+* Displaying of multimedia content using [Flash Player][wp-flash] or the
+  [Silverlight][wp-silverlight] plug-ins.
+* Running rich [Java™][wp-javaplug] applications in the browser.
+* Displaying documents in non-Web formats (PDF, [DjVu][wp-djvu]) inside
+  browser windows.
+* A number of questionable practices, like <abbr title="Virtual Private
+  Network">VPN</abbr> client software using a browser plug‑in for
+  configuration.
+
+[wp-silverlight]: https://en.wikipedia.org/wiki/Microsoft_Silverlight
+[wp-flash]: https://en.wikipedia.org/wiki/Adobe_Flash_Player
+[wp-javaplug]: https://en.wikipedia.org/wiki/Java_virtual_machine#JVM_in_the_web_browser
+[wp-djvu]: https://en.wikipedia.org/wiki/DjVu#Role_in_the_software_ecosystem
+
+
+## Why are NPAPI plug-ins being phased out?
+
+Due to the way this was done means that the browser is no longer in control
+of how the content is displayed. This not only makes it harder to style and
+do layout for them, but also it means that the browser has no control
+whatsoever over what plug-ins do—and they can do lots, given they are
+compiled, native code—, which historically has resulted in plug-ins being
+the cause for security incidents, crashes, and browser hangs.
+
+While NPAPI has served its purpose, it is a 25 year old technology and it
+shows. Web browsers' rendering engines have done big leaps and in most of
+the situations modern Web platform feature can replace plug-ins more
+efficiently and without needing to blindly trust third party code to behave
+well.
```

Comment:
Again, you started the paragraph with passive voice. That is taking a bath with socks on :-) 
Also, if you reorder of the sentences in the paragraph, you can highlight aspects that I believe
are more important.

> Today, Web browsers' rendering engines can do a better job than plug-ins, more secure and
> more effieciently. The Web platform is mature, and there is no place to blindly trust third party
> code to behave well. NPAPI is a 25 years old technology showing its age. NPAPI served its
> purpose, but it is no longer needed.

Reply:
I blame the Lating language, from both my mother tongue and my second language derive, where everything baroque and overly complicated is seen as good and sophisticated.

Thanks for untangling this one as well!